pkg/ddl/backfilling_operators.go

```diff
@@ -364,7 +364,7 @@ func (src *TableScanTaskSource) adjustStartKey(start, end kv.Key) (adjusted kv.K
 	if cpKey.Cmp(end) == 0 {
 		return cpKey, true
 	}
-	return cpKey.Next(), false
+	return cpKey, false
```

Comment:
Here is the only difference, that cause problem, am I right?

Reply:
Yes, see https://github.com/pingcap/tidb/issues/55488#issuecomment-2296771408